core/src/wsv.rs

```diff
@@ -198,6 +212,18 @@ impl<W: WorldTrait> WorldStateView<W> {
         self.new_block_notifier.subscribe()
     }
 
+    fn produce_events(&self, events: impl IntoIterator<Item = DataEvent>) {
+        let events = events.into_iter().map(Event::from);
+        let events_sender = if let Some(sender) = &self.events_sender {
+            sender
+        } else {
+            return warn!("wsv does not equip an events sender");
```

Comment:
```suggestion
            warn!("wsv does not equip an events sender");
```

Reply:
I'd like to use early return in this case